src/plugins/discover/public/application/helpers/update_search_source.ts

```diff
@@ -16,7 +16,8 @@ import { DiscoverServices } from '../../build_services';
  * Helper function to update the given searchSource before fetching/sharing/persisting
  */
 export function updateSearchSource(
-  searchSource: ISearchSource,
+  persistentSearchSource: ISearchSource,
+  volatileSearchSource: ISearchSource | undefined,
```

Comment:
:art: Would also allow us not to need to pass `undefined` explicitly to this method above.

```suggestion
  volatileSearchSource?: ISearchSource,
```

Reply:
That's not possible because the third argument is mandatory. I refactored it to just take a single object with all parameters instead.